src/index.test.mjs

```diff
@@ -65,64 +65,171 @@ test("astroRehypeRelativeMarkdownLinks", async (t) => {
+    const absolutePath = path.resolve("./fixtures/test.md");
+    const input = `<a href="${absolutePath}">foo</a>`;
+    const { value: actual } = await rehype()
+      .use(testSetupRehype)
+      .use(astroRehypeRelativeMarkdownLinks, { contentPath: "src" })
+      .process(input);
+
+    const expected = `<html><head></head><body><a href="${absolutePath}">foo</a></body></html>`;
+
+    assert.equal(actual, expected);
+  });
+
+  await t.test(
+    "should not replace absolute path if file does not exist",
+    async () => {
+      const absolutePath = `${path.dirname("./fixtures/test.md")}/does-not-exist.md`;
+      const input = `<a href="${absolutePath}">foo</a>`;
+      const { value: actual } = await rehype()
+        .use(testSetupRehype)
+        .use(astroRehypeRelativeMarkdownLinks, { contentPath: "src" })
+        .process(input);
+
+      const expected = `<html><head></head><body><a href="${absolutePath}">foo</a></body></html>`;
+
+      assert.equal(actual, expected);
+    }
+  );
+
+  await t.test(
+    "should not transform index.md file paths if file does not exist",
+    async () => {
+      const input = '<a href="./fixtures/does-not-exist/index.md">foo</a>';
+      const { value: actual } = await rehype()
+        .use(testSetupRehype)
+        .use(astroRehypeRelativeMarkdownLinks, { contentPath: "src" })
+        .process(input);
+
+      const expected =
+        '<html><head></head><body><a href="./fixtures/does-not-exist/index.md">foo</a></body></html>';
+
+      assert.equal(actual, expected);
+    }
+  );
+
+  await t.test("should not replace path if .md directory exists", async () => {
+    const input = '<a href="./fixtures/dir-exists.md">foo</a>';
     const { value: actual } = await rehype()
       .use(testSetupRehype)
       .use(astroRehypeRelativeMarkdownLinks, { contentPath: "src" })
       .process(input);
 
     const expected =
-      '<html><head></head><body><a href="/fixtures/test?q=q#hash">foo</a></body></html>';
+      '<html><head></head><body><a href="./fixtures/dir-exists.md">foo</a></body></html>';
 
     assert.equal(actual, expected);
   });
 
-  await t.test("should prefix base to output", async () => {
-    const input = '<a href="./fixtures/test.md">foo</a>';
+  await t.test("should not replace path if .mdx directory exists", async () => {
+    const input = '<a href="./fixtures/dir-exists.mdx">foo</a>';
     const { value: actual } = await rehype()
       .use(testSetupRehype)
-      .use(astroRehypeRelativeMarkdownLinks, {
-        contentPath: "src",
-        basePath: "/testBase",
-      })
+      .use(astroRehypeRelativeMarkdownLinks, { contentPath: "src" })
       .process(input);
 
     const expected =
-      '<html><head></head><body><a href="/testBase/fixtures/test">foo</a></body></html>';
+      '<html><head></head><body><a href="./fixtures/dir-exists.mdx">foo</a></body></html>';
 
     assert.equal(actual, expected);
   });
```

Comment:
For these, I'm wondering if we should transform them as the directory exists, even though an index-esque file doesn't exist. But, I thinks it's likely fine as it would be new functionality. Not sure if it would match what Astro is doing